src/test/java/edu/ucsb/cs156/frontiers/security/CourseSecurityTests.java

```diff
@@ -0,0 +1,124 @@
+package edu.ucsb.cs156.frontiers.security;
+
+import edu.ucsb.cs156.frontiers.config.CourseSecurity;
+import edu.ucsb.cs156.frontiers.entities.Course;
+import edu.ucsb.cs156.frontiers.entities.User;
+import edu.ucsb.cs156.frontiers.errors.EntityNotFoundException;
+import edu.ucsb.cs156.frontiers.models.CurrentUser;
+import edu.ucsb.cs156.frontiers.repositories.CourseRepository;
+import edu.ucsb.cs156.frontiers.repositories.UserRepository;
+import edu.ucsb.cs156.frontiers.services.CurrentUserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.context.annotation.Import;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.test.context.support.TestExecutionEvent;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.NestedTestConfiguration;
+import org.springframework.test.context.bean.override.mockito.MockitoBean;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration
+@NestedTestConfiguration(NestedTestConfiguration.EnclosingConfiguration.INHERIT)
+@Import({CourseSecurity.class, DummyCourseSecurity.class})
+@EnableMethodSecurity
+public class CourseSecurityTests {
+
+    @MockitoBean
+    private CourseRepository courseRepository;
+
+    @MockitoBean
+    CurrentUserService currentUserService;
+
+    @MockitoBean
+    UserRepository userRepository;
+
+    @Autowired
+    DummyCourseSecurity DummyCourseSecurity;
+
+
+    @Nested

```

Comment:
What does the `@Nested` do here?

I think we might want to add an article on the ucsb-cs156 topics pages about this.